File: java-workshop/part-1-powermock/src/main/java/service/PricingService.java
package service;

import domain.OrderItem;

import java.util.List;

/**
 * Pricing service with constants.
 */
public class PricingService {
    public static final double TAX_RATE = 0.20;
    public static final double FREE_SHIPPING_THRESHOLD = 100.0;
    public static final double SHIPPING_COST = 5.99;

    public double calculateTotal(List<OrderItem> items) {
        double subtotal = calculateSubtotal(items);
        double tax = calculateTax(subtotal);
        double shipping = calculateShipping(subtotal);
        return subtotal + tax + shipping;
    }

    private double calculateSubtotal(List<OrderItem> items) {
        return items.stream()
            .mapToDouble(item -> item.quantity() * item.unitPrice())
            .sum();
    }

    private double calculateTax(double subtotal) {
        return subtotal * TAX_RATE;
    }

    private double calculateShipping(double subtotal) {
        return subtotal >= FREE_SHIPPING_THRESHOLD ? 0.0 : SHIPPING_COST;
    }
}

File: kotlin-workshop/part-1-powermock/src/main/kotlin/service/PricingService.kt
package service

import domain.OrderItem

/**
 * Pricing service - relatively simple, but still has internal logic
 */
class PricingService {
    
    companion object {
        const val TAX_RATE = 0.20
        const val FREE_SHIPPING_THRESHOLD = 100.0
        const val SHIPPING_COST = 5.99
    }
    
    fun calculateTotal(items: List<OrderItem>): Double {
        val subtotal = calculateSubtotal(items)
        val tax = calculateTax(subtotal)
        val shipping = calculateShipping(subtotal)
        return subtotal + tax + shipping
    }
    
    private fun calculateSubtotal(items: List<OrderItem>): Double {
        return items.sumOf { it.quantity * it.unitPrice }
    }
    
    private fun calculateTax(subtotal: Double): Double {
        return subtotal * TAX_RATE
    }
    
    private fun calculateShipping(subtotal: Double): Double {
        return if (subtotal >= FREE_SHIPPING_THRESHOLD) 0.0 else SHIPPING_COST
    }
}

